### PR TITLE
Bump to PHPStan 0.12

### LINKED ISSUE #91: Ensure PHPStan 0.12 support
😬 let's see if PHPStan 0.12 works with this extension.

### PR DESCRIPTION
**Pull request by mglaman**
Fixes #91 

- [x] Need to test schema of extension.neon to resolve `Unexpected option 'parameters › drupalServiceMap', 'parameters › entityTypeStorageMapping'.`
- [x] Configurations for testing PHPStan analysis rules are not running 😬 
- [ ] Document breaking config changes (no more root `drupal:`, under `parameters`
- [x] ServiceDynamicReturnTypeExtension is broken

### CODE REVIEW

File: tests/fixtures/config/phpunit-drupal-phpstan.neon [hunk 1]
```diff
@@ -4,9 +4,8 @@ parameters:
 	# Ignore functions for the PHPUnit bootstrap process.
 	ignoreErrors:
 		- '#Function drupal_phpunit_[a-zA-Z0-9\\_]+ not found#'
-drupal:
-	drupal_root: %currentWorkingDirectory%
+	drupal:
+		drupal_root: %currentWorkingDirectory%
 includes:
 	- ../../../extension.neon
-	- ../../../vendor/phpstan/phpstan/conf/config.level2.neon
```
Comment (mglaman) [inline tests/fixtures/config/phpunit-drupal-phpstan.neon]: Need to get this back, somehow.

**Comment by mglaman** [comment]
Okay. So this works when I run the test with Xdebug. Otherwise it fails. 🤷‍♂

File: extension.neon [hunk 2]
```diff
@@ -9,25 +11,26 @@ parameters:
...
-		taxonomy_term: Drupal\taxonomy\TermStorage
-		user: Drupal\user\UserStorage
+	drupal:
+		entityTypeStorageMapping:
+			node: Drupal\node\NodeStorage
+			taxonomy_term: Drupal\taxonomy\TermStorage
+			user: Drupal\user\UserStorage
+parametersSchema:
+	drupal: structure([
+		drupal_root: string()
```
Comment (mglaman) [inline extension.neon]: > The mandatory option 'parameters › drupal › drupal_root' is missing.

File: README.md [hunk 3]
```diff
@@ -83,18 +83,20 @@ includes:
 
 By default, the PHPStan Drupal extension will try to determine your Drupal project's root directory based on the working
 directory that PHPStan is checking. If this is not working properly, you can explicitly define the Drupal project's root
-directory using the `drupal_root` parameter.
+directory using the `drupal.root` parameter.
```
Comment (dan2k3k4) [inline README.md]: I believe this should be changed to: `drupal.drupal_root`

**Comment by ondrejmirtes** [comment]
I've sent you a PR with a couple of minor things :) https://github.com/mglaman/phpstan-drupal/pull/99

The first commit really makes the extension work - without it, user wouldn't be able to use the extension without defining `drupal_root` parameter.

**Pull request merged**